src/base/static/components/input-form/index.js

```diff
@@ -56,40 +56,42 @@ class InputForm extends Component {
       nextProps.selectedCategory !== this.props.selectedCategory
     ) {
       this.initializeForm(nextProps.selectedCategory);
+      this.setState(prevState => ({
+        fields: this.getNewFields(prevState.fields),
+        isFormSubmitting: false,
+        isMapPositioned: false,
+        formValidationErrors: new Set(),
+        showValidityStatus: false,
+      }));
     }
   }
```

Comment:
It might be out of scope for this PR, but `componentWillReceiveProps` is getting deprecated in future version of React, and I'm not sure if we need this lifecycle method here. See if you can use something like `componentDidUpdate(prevProps)` instead.

ie:
```
  componentDidUpdate(prevProps) {
    if (prevProps.selectedCategory !== this.props.selectedCategory) {
      this.initializeForm(this.props.selectedCategory)
      // etc
    }
  }

```

Reply:
Oh no, I like `componentWillReceiveProps` :(. But yeah, we should be able to replace it with `componentDidUpdate`. I'd say we should do that in another PR, since we use `componentWillReceiveProps` in several places and we might want to refactor all of them in one go.